### PR TITLE
[SDL 0200] - Removing url parameter max length

### LINKED ISSUE #1110: [SDL 0200] - Removing URL Parameter Max Length
Proposal: [Removing URL Parameter Max Length](https://github.com/smartdevicelink/sdl_evolution/blob/master/proposals/0200-Removing-URL-Param-Max-Length.md)

> This proposal is to remove maximum length attribute for URL strings provided to OnSystemRequest, StartStream and StartAudioStream and let the URL strings be of any length permissible by the system architecture.

Review: https://github.com/smartdevicelink/sdl_evolution/issues/603

Steering Committee Decision:

> The Steering Committee accepted this proposal on 10/16/2018. The following was discussed in the Steering Committee meeting on 2018-10-16, with regards to this proposal: Although there are no iOS or Android limitations over the size and length of URLs, excessively long ones may take a long time to create / parse, and extensive data should be sent via a method that is not a URL (e.g. JSON in the bulk data).
Lastly, while it's understood that SDLC members cannot share all use cases for features publicly, it was acknowledged that in the future, when requesting specific features, it is helpful to have real world examples to better understand the reasoning and planned implementation for the requested changes.

### PR DESCRIPTION
**Pull request by yoooriii**
Fixes #1110 

This PR is **ready** for review.

### Risk
This PR makes **no** API changes.

### Summary
This PR doesn't provide any library changes. Still, it gives unit tests that cover OnSystemRequest class itself with additional checking of the url parameter to be sure it is not changed in the library over the set/get process.

##### Enhancements
New unit tests added for OnSystemRequest class

### CLA
- [x] I have signed [the CLA](https://docs.google.com/forms/d/e/1FAIpQLSdsgJY33VByaX482zHzi-xUm49JNnmuJOyAM6uegPQ2LXYVfA/viewform)

### CODE REVIEW

**Comment by yoooriii** [comment]
Hello. @jordynmackool  Could you please review this PR?

**Comment by jordynmackool** [comment]
@yoooriii Please tag @theresalech once it has been reviewed and approved by the Ford team (@atiwari9).

**Comment by atiwari9** [comment]
+ @SatbirTanda

File: SmartDeviceLinkTests/RPCSpecs/NotificationSpecs/SDLOnSystemRequestSpec.m [hunk 1]
```diff
@@ -74,6 +74,45 @@
...
+        NSString *testURL = @"";
+        testRequest.url = testURL;
+        expect(testRequest.url).to(equal(testURL));
+    });
+
+    it(@"Should set and get url correctly (1000 sym long url)", ^ {
+        const int size = 1000;
+        SDLOnSystemRequest* testRequest = [[SDLOnSystemRequest alloc] init];
+        NSMutableString *testURL = [NSMutableString stringWithCapacity:1024];
+        [testURL setString:@"www.google.com"];
```
Comment (SatbirTanda) [inline SmartDeviceLinkTests/RPCSpecs/NotificationSpecs/SDLOnSystemRequestSpec.m:89]: use `https://www.smartdevicelink.com`

File: SmartDeviceLinkTests/RPCSpecs/NotificationSpecs/SDLOnSystemRequestSpec.m [hunk 2]
```diff
@@ -74,6 +74,45 @@
...
+        }
+        testRequest.url = testURL;
+        expect(testRequest.url).to(equal(testURL));
+    });
+
+    it(@"Should set and get url correctly (extra long url)", ^ {
+        const int size = 1024 * 1024;
+        SDLOnSystemRequest* testRequest = [[SDLOnSystemRequest alloc] init];
+        NSMutableString *testURL = [NSMutableString stringWithCapacity:size + 20];
+        [testURL setString:@"www.google.com"];
```
Comment (SatbirTanda) [inline SmartDeviceLinkTests/RPCSpecs/NotificationSpecs/SDLOnSystemRequestSpec.m:105]: use `https://www.smartdevicelink.com`

**Comment by yoooriii** [comment]
Hello,  @theresalech could you please review?

File: SmartDeviceLinkTests/RPCSpecs/NotificationSpecs/SDLOnSystemRequestSpec.m [hunk 3]
```diff
@@ -14,6 +14,7 @@
 #import "SDLOnSystemRequest.h"
 #import "SDLRequestType.h"
 
+static NSString * const SDLTestURL = @"https://www.smartdevicelink.com";
```
Comment (joeljfischer) [inline SmartDeviceLinkTests/RPCSpecs/NotificationSpecs/SDLOnSystemRequestSpec.m:17]:
```suggestion
static NSString *const SDLTestURL = @"https://www.smartdevicelink.com";
```

File: SmartDeviceLinkTests/RPCSpecs/NotificationSpecs/SDLOnSystemRequestSpec.m [hunk 4]
```diff
@@ -56,7 +57,7 @@
         
         expect(testNotification.requestType).to(equal(SDLRequestTypeFileResume));
         expect(testNotification.requestSubType).to(equal(@"subtype"));
-        expect(testNotification.url).to(equal([@"www.google.com" mutableCopy]));
+        expect(testNotification.url).to(equal([SDLTestURL mutableCopy]));
```
Comment (joeljfischer) [inline SmartDeviceLinkTests/RPCSpecs/NotificationSpecs/SDLOnSystemRequestSpec.m:60]:
```suggestion
        expect(testNotification.url).to(equal(SDLTestURL));
```

File: SmartDeviceLinkTests/RPCSpecs/NotificationSpecs/SDLOnSystemRequestSpec.m [hunk 5]
```diff
@@ -74,6 +75,45 @@
         expect(testNotification.offset).to(beNil());
         expect(testNotification.length).to(beNil());
     });
+
+    it(@"Should set and get url correctly (zero length url)", ^ {
```
Comment (joeljfischer) [inline SmartDeviceLinkTests/RPCSpecs/NotificationSpecs/SDLOnSystemRequestSpec.m:79]:
```suggestion
    it(@"Should set and get url correctly (zero length url)", ^{
```

File: SmartDeviceLinkTests/RPCSpecs/NotificationSpecs/SDLOnSystemRequestSpec.m [hunk 6]
```diff
@@ -74,6 +75,45 @@
...
+            const NSRange range = NSMakeRange(size, testURL.length - size);
+            [testURL deleteCharactersInRange:range];
+        }
+        testRequest.url = testURL;
+        expect(testRequest.url).to(equal(testURL));
+    });
+
+    it(@"Should set and get url correctly (extra long url)", ^ {
+        const int size = 1024 * 1024;
+        SDLOnSystemRequest* testRequest = [[SDLOnSystemRequest alloc] init];
```
Comment (joeljfischer) [inline SmartDeviceLinkTests/RPCSpecs/NotificationSpecs/SDLOnSystemRequestSpec.m:104]:
```suggestion
        SDLOnSystemRequest *testRequest = [[SDLOnSystemRequest alloc] init];
```

File: SmartDeviceLinkTests/RPCSpecs/NotificationSpecs/SDLOnSystemRequestSpec.m [hunk 7]
```diff
@@ -74,6 +75,45 @@
         expect(testNotification.offset).to(beNil());
         expect(testNotification.length).to(beNil());
     });
+
+    it(@"Should set and get url correctly (zero length url)", ^ {
+        SDLOnSystemRequest* testRequest = [[SDLOnSystemRequest alloc] init];
+        NSString *testURL = @"";
+        testRequest.url = testURL;
+        expect(testRequest.url).to(equal(testURL));
```
Comment (joeljfischer) [inline SmartDeviceLinkTests/RPCSpecs/NotificationSpecs/SDLOnSystemRequestSpec.m:83]: Also test what happens when turning this string into an `NSURL`

File: SmartDeviceLinkTests/RPCSpecs/NotificationSpecs/SDLOnSystemRequestSpec.m [hunk 8]
```diff
@@ -74,6 +75,45 @@
...
+        [testURL setString:SDLTestURL];
+        while(size > testURL.length) {
+            [testURL appendString:@"/testcomponent"];
+        }
+        if (size < testURL.length) {
+            const NSRange range = NSMakeRange(size, testURL.length - size);
+            [testURL deleteCharactersInRange:range];
+        }
+        testRequest.url = testURL;
+        expect(testRequest.url).to(equal(testURL));
```
Comment (joeljfischer) [inline SmartDeviceLinkTests/RPCSpecs/NotificationSpecs/SDLOnSystemRequestSpec.m:99]: Also test what happens when turning this string into an `NSURL` (it should create one)

File: SmartDeviceLinkTests/RPCSpecs/NotificationSpecs/SDLOnSystemRequestSpec.m [hunk 9]
```diff
@@ -74,6 +75,45 @@
...
```
Comment (joeljfischer) [inline SmartDeviceLinkTests/RPCSpecs/NotificationSpecs/SDLOnSystemRequestSpec.m:115]: Also test what happens when turning this string into a URL (it should create one)

**Comment by yoooriii** [comment]
Hello. I have updated the PR using the proposed changes, @joeljfischer could you please review?

**Pull request merged**